Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsBuildProvider.cs

```diff
@@ -25,8 +25,12 @@ public void OnPreprocessBuild(BuildReport report)
 
 #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
             // Determine if we need to preload project-wide InputActionsAsset.
+            // Also make sure we set project wide flag on object that gets serialized into build.
+            // It will allow us to find the right object if multiple preloaded assets
+            // Input System asset + user assets.
             var actions = InputSystem.actions;
             var actionsMissing = NeedsToBeAdded(preloadedAssets, actions, ref newSize);
+            actions.m_IsProjectWide = true; 
```

Comment:
Should we be making sure to set this and remove this each time an asset is assigned to the 'actions' in general ?

I'm just not sure if this could end up persisting on the asset after its unassigned from actions.

Reply:
I did exactly that when I put this together and then I realised that we only bundle a single preloaded asset anyway so it doesn't matter. All other assets are accessed from either disk/json and when deserialised/imported that way that field will always be false (not mapped from json). If this would be set on an InputActionAsset object as part of a build and then the object is swapped to another you are right that both of them might have the flag set in Unity default serialisation mechanism. But there wouldn't be any harm since the only access to this is internally and in the Resource.LoadAllResources during initialisation which will only retrieve preloaded assets. Hence I believe the only risk would be that a user first create an asset and use it as project-wide within a build. Then switches it, then add their own BuildSettingsProvider to inject it as a preloaded asset. I guess what we could do would be to add into InputActionAsset OnValidate function to set it to false if not equal to InputSystem.asset to make it even harder if you believe its necessary?

Also note that I decided to leave InputSettings alone (still uses the brittle approach which it has since for as long as it has existed), wonder if that should be filed as a separate future ticket to correct with same pattern.

Reply:
And to answer your question, it could until its reimported I believe.

Reply:
I did the change you requested but wonder if we should really do it that way since only build related.